modules/vertx-swagger-codegen/src/main/resources/javaVertXServer/apiVerticle.mustache

```diff
@@ -33,11 +33,22 @@ public class {{classname}}Verticle extends AbstractVerticle {
                 {{/isString}}{{^isString}}{{{dataType}}} {{paramName}} = Json.mapper.readValue(message.body().getString("{{baseName}}"), {{{dataType}}}.class);
                 {{/isString}}{{/isPrimitiveType}}{{^isPrimitiveType}}{{{dataType}}} {{paramName}} = Json.mapper.readValue(message.body().getJsonObject("{{baseName}}").encode(), {{{dataType}}}.class);
                 {{/isPrimitiveType}}{{/isListContainer}}{{/allParams}}{{/hasParams}}
-                {{#returnType}}{{{returnType}}} result = service.{{operationId}}({{#hasParams}}{{#allParams}}{{paramName}}{{#hasMore}}, {{/hasMore}}{{/allParams}}{{/hasParams}});
-                {{#isListContainer}}message.reply(new JsonArray(Json.encode(result)).encodePrettily());{{/isListContainer}}{{^isListContainer}}message.reply(new JsonObject(Json.encode(result)).encodePrettily());{{/isListContainer}}{{/returnType}}{{^returnType}}service.{{operationId}}({{#hasParams}}{{#allParams}}{{paramName}}{{#hasMore}}, {{/hasMore}}{{/allParams}}{{/hasParams}});
-                message.reply(null);{{/returnType}}
-            } catch ({{classname}}Exception e) {
-                message.fail(e.getStatusCode(), e.getStatusMessage());
+                service.{{operationId}}({{#hasParams}}{{#allParams}}{{paramName}}{{#hasMore}}, {{/hasMore}}{{^hasMore}}, {{/hasMore}}{{/allParams}}{{/hasParams}}result -> {
+                    if (result.succeeded())
+                        {{#isListContainer}}message.reply(new JsonArray(Json.encode(result.result())).encodePrettily());{{/isListContainer}}{{^isListContainer}}message.reply(new JsonObject(Json.encode(result.result())).encodePrettily());{{/isListContainer}}
+                    else {
+                        Throwable cause = result.cause();
```

Comment:
I thought about it, and decided against it, as non-200 results are pretty common, especially the 4xx codes which are caused by invalid client requests. For example, a pretty common error is a failing authentication (401) because of e.g. a token becoming invalid. I recommend not logging errors for those, as it will clutter up the log. What do you think?